### ISSUE TITLE
Add a core example of using Tkinter with Twisted

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5631]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)| @jesstess reported|
|-|-|
|Trac ID|trac#5631|
|Type|enhancement|
|Created|2012-04-10 01:58:58Z|

We talk about how to integrate Tkinter in http://twistedmatrix.com/documents/current/core/howto/choosing-reactor.html#auto15, but we have no explicit, basic example in doc/core/examples.

We do already have a PyUI demo (doc/core/examples/pyuidemo.py).

Attachments:

* [tkinterdemo.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/fc7/fc7e790f8b6988f6c67ee401b60604fa53cdb187/4a5f0300258a790ee87e1e534dda26c0c33c2196.patch) (1347 bytes) - added by lesleylam on 2012-04-12 18:37:40Z - Example of integrating Tkinter with Twisted

<details><summary>Searchable metadata</summary>

```
trac-id__5631 5631
type__enhancement enhancement
reporter__jesstess jesstess
priority__normal normal
milestone__ 
branch__ 
branch_author__lesleylam lesleylam
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation_easy documentation easy
time__1334023138000000 1334023138000000
changetime__1334402072000000 1334402072000000
version__None None
owner__itamar itamar
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="lesleylam's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|lesleylam commented|
|-|-|

I've written up a basic example for integrating Tkinter with Twisted.  Attached is the patch file for review.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @itamarst|
|-|-|

Thanks for the contribution - looks good! 

I will make three minor tweaks and then commit:

1. If possible, should avoid using "from module import *", it fills up the module with unexpected names and makes it harder to figure out where objects came from. So I'll be doing "from Tkinter import Frame, Button" etc.. Apparently some GUI frameworks recommend * imports because they have so many objects, though.

2. I'll rephrase the news file slightly.

3. The file seemed to have mixed tabs and spaces, which can be problematic.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set status to `closed`|
|-|-|

(In [34214]) Merge patch: Tkinter/Twisted integration example.

Author: lesleylam
Review: itamar
Fixes: [#5631](https://github.com/twisted/twisted/issues/5631)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

The example wasn't included in the index.xhtml.
